tests/test_replicate.py

```diff
@@ -268,16 +326,124 @@ def test_call_streamming_method_called():
     lock = manager.Lock()
 
     job_info = scripts.aws_replicate.JobInfo(
-            {},
-            gen_mock_manifest_data()[0],
-            1,
-            1,
-            "",
-            copied_objects,
-            source_objects,
-            manager_ns,
-            "bucket",
-        )
+        {},
+        gen_mock_manifest_data()[0],
+        1,
+        1,
+        "",
+        copied_objects,
+        source_objects,
+        manager_ns,
+        "bucket",
+    )
     scripts.aws_replicate.exec_aws_copy(lock, False, job_info)
     assert subprocess.Popen.call_count == 0
     assert scripts.aws_replicate.stream_object_from_gdc_api.call_count == 1
+
+
+def test_remove_changed_url():
+    """
+    Test removing changed url for indexd
+    :return:
+    """
+    urls = ["s3://tcga-open/uuid1/filname1", "gs://gdc-tcga-open/uuid1/filname1"]
+    urls_metadata = {
+        "s3://tcga-open/uuid1/filname1": {},
+        "gs://gdc-tcga-open/uuid1/filname1": {},
+    }
+    url = "s3://tcga-controlled/uuid1/filname1"
+
+    doc = MockIndexDRecord(urls=urls, urls_metadata=urls_metadata)
+    doc, modified = indexd_utils._remove_changed_url(doc, url)
```

Comment:
This function assumes files are only at a single location in s3 and at a single location in google, right? If you want to add the new URL `s3//.....` it removes the old s3 URL? and then adding the new URL is handled outside of this function

if you have time would you mind adding more details to the docstring here? https://github.com/uc-cdis/dcf-dataservice/blob/master/scripts/indexd_utils.py#L20

Reply:
They can have more than single location in both s3 and gs. The removing one is the one pointing to the bucket for the same project but different acl